core/indexer/dataDispatcher.go

```diff
@@ -51,35 +61,78 @@ func (d *dataDispatcher) startWorker(ctx context.Context) {
 	for {
 		select {
 		case <-ctx.Done():
-			log.Debug("dispatcher's go routine is stopping...")
+			d.stopWorked()
 			return
 		case wi := <-d.chanWorkItems:
-			d.doWork(wi)
+			timeout := d.doWork(wi)
+			if timeout {
+				d.stopWorked()
+				return
+			}
 		}
 	}
 }
 
+func (d *dataDispatcher) stopWorked() {
```

Comment:
rename to `stopWork` or something similar.

Reply:
done.